packages/untp-test-suite/src/core/processTestSuite.ts

```diff
@@ -1,39 +1,84 @@
-import path from 'path';
-import { ConfigCredentials, TestSuite, TestSuiteResult } from './types/index';
+import {
+  ConfigContent,
+  ConfigCredentials,
+  IValidatedCredentials,
+  ICredentialTestResult,
+  IFinalReport,
+  TemplateName,
+  TestSuite,
+} from './types/index.js';
 import { readJsonFile, validateCredentialConfigs } from './utils/common.js';
 import { dynamicLoadingSchemaService } from './services/dynamic-loading-schemas/loadingSchema.service.js';
 import { hasErrors } from './services/json-schema/validator.service.js';
+import { templateMapper } from '../templates/mapper.js';
+import { generateFinalMessage, getTemplateName } from '../interfaces/utils/common.js';
+
+const processCheckDataBySchema = async (credentialConfig: ConfigContent): Promise<IValidatedCredentials> => {
+  const { type, version, dataPath } = credentialConfig;
+  const [schema, data] = await Promise.all([dynamicLoadingSchemaService(type, version), readJsonFile(dataPath)]);
+
+  const errors = hasErrors(schema, data);
+
+  return {
+    ...credentialConfig,
+    errors,
+  };
+};
+
+const generateTestSuiteResultByTemplate = async (
+  testSuiteResultAjv: IValidatedCredentials[],
+): Promise<ICredentialTestResult[]> => {
+  const credentialsTemplate = await Promise.all(
+    testSuiteResultAjv.map(async (value) => {
+      const templateName = getTemplateName(value);
+      const message = await templateMapper(templateName, value);
+
+      return JSON.parse(message);
+    }),
+  );
+
+  return credentialsTemplate;
+};
 
 /**
  * Process the test suite
  * @param credentialConfigsPath - The path to the credential configs
  * @returns The test suite result
  */
 export const processTestSuite: TestSuite = async (credentialConfigsPath) => {
-  const credentialConfigs = await readJsonFile<ConfigCredentials>(credentialConfigsPath);
-  const validateCredentialResult = validateCredentialConfigs(credentialConfigsPath, credentialConfigs.credentials);
+  try {
+    // Read and validate credential configs
+    const credentialConfigs = await readJsonFile<ConfigCredentials>(credentialConfigsPath);
+    const validateCredentialResult = validateCredentialConfigs(credentialConfigsPath, credentialConfigs.credentials);
 
-  const configContentNoError = validateCredentialResult.filter((config) => config.errors === null);
+    // Filter out configs with no errors
+    const configContentNoError = validateCredentialResult.filter((config) => config.errors === null);
 
```

Comment:
I think the `configContentNoError` variable here always contains an empty array. This is because the `config` parameter of the callback function inside this `filter` function never has an `errors` field with 'null' data.

You can review the details of the `validateCredentialConfigs` function (it's invoked at line 52), where you'll notice that the `errors` field is always an array.